feat(scheduler): Allow configured resource groups to bypass cluster-overload throttling (#27642) (#27642)
prestodb/presto: presto-docs/src/main/sphinx/admin/properties.rst

The corresponding session property is :ref:`admin/properties-session:\`\`materialized_view_stale_read_behavior\`\``.

Cluster Overload Properties

> ----------------------------

Reviewer: ```suggestion
---------------------------
```
The formatting line should be exactly the same number of characters as the line being formatted.

* **Type:** ``integer``
* **Default value:** ``0``

Maximum number of workers that may be in an overloaded state before the cluster is
considered overloaded. Only used when ``cluster-overload.overload-policy-type`` is set
to ``overload_worker_cnt_based_throttling``.

``cluster-overload.allowed-overload-workers-pct``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``double``
* **Default value:** ``0.01``

Maximum fraction of workers (between 0 and 1) that may be in an overloaded state
before the cluster is considered overloaded. Only used when
``cluster-overload.overload-policy-type`` is set to
``overload_worker_pct_based_throttling``.

``cluster.overload-check-cache-ttl-secs``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``
* **Default value:** ``5``

How frequently (in seconds) the coordinator re-evaluates the cluster overload state.
Between checks, the most recently computed state is cached and reused.

``cluster-overload.bypass-resource-groups``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``string``
* **Default value:** (empty)

Comma-separated list of fully-qualified resource group ids (dot-separated segments,

> e.g. ``global.admin, global.etl.priority``) that bypass cluster-overload throttling.

Reviewer: ```suggestion
such as ``global.admin, global.etl.priority``) that bypass cluster-overload throttling.
```
Avoid Latin abbreviations. See the [GitLab documentation style guide recommended word list entry for e.g.](https://docs.gitlab.com/development/documentation/styleguide/word_list/#eg) for discussion and alternatives.

Listed groups are still subject to all other admission gates — per-group concurrency,
memory, and CPU limits continue to apply.

A resource group bypasses throttling when it lies on the same root-to-leaf path as any
configured entry: equal to a listed group, an ancestor of one, or a descendant of one.
Sibling groups that do not appear in the list remain throttled. Path-based matching is
required because the admission check runs at every ancestor in the resource group
hierarchy; without it a non-listed ancestor would veto a listed leaf group.

The bypass set is snapshotted at coordinator startup; changes require a coordinator
restart. A ``throttlingBypassCount`` JMX counter on ``ClusterResourceChecker`` tracks
how many times the bypass was applied.

> Malformed entries (e.g. ``global..admin``) cause a startup failure.

Reviewer: ```suggestion
Malformed entries (for example, ``global..admin``) cause a startup failure.
```
